src/common/config_validator.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "3rd_party/yaml-cpp/yaml.h"
+#include "common/config_parser.h"
+
+namespace marian {
+
+class ConfigValidator
```

Comment:
comment what this is, and what it is meant to be used for

Reply:
That's an ugly class created from code pieces extracted from the old config parser (slightly re-organized) to make sure that removing boost::program_options doesn't break too much. Will be changed in the upcoming refactorization of CLI.

Added TODO.